### PR TITLE
Bump SemanticKernel dependencies to v1.6.1

### PR DESCRIPTION
**Pull request by marcominerva**
## Motivation and Context (Why the change? What's the scenario?)
This PR solves a version mismatch between libraries. See https://github.com/microsoft/kernel-memory/issues/361.

## High level description (Approach, Design)
This PR updates the package dependencies to correctly support the latest version of **Azure.AI.OpenAI library**, that is required to use the right API version of the service.

### CODE REVIEW

File: service/Core/Handlers/DependencyInjection.cs [hunk 1]
```diff
@@ -103,8 +103,8 @@ public static IServiceCollection AddDefaultHandlersAsHostedServices(this IServic
         // Build generic type: HandlerAsAHostedService<THandler>
         Type handlerAsAHostedServiceTHandler = typeof(HandlerAsAHostedService<>).MakeGenericType(tHandler);
 
-        Func<IServiceProvider, IHostedService> implementationFactory =
-            serviceProvider => (IHostedService)ActivatorUtilities.CreateInstance(serviceProvider, handlerAsAHostedServiceTHandler, stepName);
+        IHostedService implementationFactory(IServiceProvider serviceProvider)
```
Comment (dluc) [inline service/Core/Handlers/DependencyInjection.cs:106]: why this change?

Comment (marcominerva) [reply service/Core/Handlers/DependencyInjection.cs:106]: Because trying to rebuild the solution (using **KernelMemory.sln**, `main` branch), I get this compilation error:

image

And the message point me to this page: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0039. I just updated Visual Studio 2022 to v17.9.3, perhaps it is a new analyzer?

Comment (dluc) [reply service/Core/Handlers/DependencyInjection.cs:106]: looks like a new analyzer only in VS, odd

**Review by dluc — APPROVED** [review]
